### PR TITLE
Add validation for the CN prefix in key pair creation

### PR DESCRIPTION
**Pull request by marians**
For https://github.com/giantswarm/giantswarm/issues/4275

### Preview

cn-prefix-validation

### CODE REVIEW

File: src/components/organizations/cluster_key_pairs.js [hunk 1]
```diff
@@ -51,8 +52,18 @@ class ClusterKeyPairs extends React.Component {
   }
 
   handleCNPrefixChange(e) {
+    var error = null;
+    if (e.target.value !== '') {
+      var regex = /^[a-z0-9][a-z0-9-]*[a-z0-9]$/g;
+      var found = e.target.value.match(regex);
+      if (found === null) {
+        error = 'Please use characters a-z 0-9 and -';
+      }
```
Comment (pipo02mix) [inline src/components/organizations/cluster_key_pairs.js]: Nitpick 
`Please use characters a-z, 0-9 or -`

File: src/styles/components/_form.sass [hunk 2]
```diff
@@ -26,6 +26,8 @@ form
   .text-field-hint
     font-size: 12px
     color: #ccc
+    .error
+      color: #771d1d
```
Comment (pipo02mix) [inline src/styles/components/_form.sass]: To me the contrast is not fine. I would select another color tone which highlights from the background

**Comment by marians** [comment]
Totally agreeing. What was I thinking?

Unfortunately, red on blue doesn't work very well in any case. And I don't want to make it pink. So I am using the "gold" color we use for the "upgrade available" notification. WDYT?

image

**Comment by marians** [comment]
Contrast is still not very good. This is also because we use a lighter background in modals.

Do we need that lighter background, especially as we place it above a dark glass pane?

Here is how modals would look with our standard background color:

image

This is how they look currently:

image

**Comment by marians** [comment]
Opinions? Should we make the modal darker?

**Comment by oponder** [comment]
I think the gold on the lighter modal looks ok!

Darker modal is also ok, but need to do a bit of exploratory testing and make sure it looks ok for other modals too.

**Comment by pipo02mix** [comment]
Agree with Oli

File: src/components/organizations/cluster_key_pairs.js (hunk 1 above)
Comment (marians) [reply src/components/organizations/cluster_key_pairs.js]: done

File: src/styles/components/_form.sass (hunk 2 above)
Comment (marians) [reply src/styles/components/_form.sass]: I settled for the darker blue in the modal

**Comment by marians** [comment]
I changed the modal background to the darker blue and clicked through the other dialogs. Looks good to me. Actually this bring more contrast to the modals over all, which is a good thing I think.

**Pull request merged**